### PR TITLE
[dpdk] shorten label names

### PR DESCRIPTION
**Pull request by kamleshbhalui**

### CODE REVIEW

File: backends/dpdk/dpdkAsmOpt.h [hunk 1]
```diff
@@ -280,15 +280,25 @@ class ShortenTokenLength : public Transform {
...
         return ls;
     }
 
-    const IR::Node* preorder(IR::DpdkApplyStatement *as) override{
+    const IR::Node* preorder(IR::DpdkApplyStatement *as) override {
         as->table = shortenString(as->table);
         return as;
     }
+
+    const IR::Node* preorder(IR::DpdkJmpStatement *j) override {
```
Comment (usha1830) [inline backends/dpdk/dpdkAsmOpt.h:293]: We are checking somewhere else that the def and use are matching for shortened names, right?

Comment (kamleshbhalui) [reply backends/dpdk/dpdkAsmOpt.h:293]: JmpStatement(uses) and LabelStatement (def)is guaranteed.

**Review by usha1830 — APPROVED** [review]
Looks fine now.

File: testdata/p4_16_samples_outputs/pna-too-big-label-name-dpdk.p4.spec [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+struct main_metadata_t {
+	bit<32> pna_main_input_metadata_direction
+	bit<32> pna_main_input_metadata_input_port
+	bit<32> pna_main_output_metadata_output_port
+	bit<32> MainControlT_tmpDir
+}
+metadata instanceof main_metadata_t
+
+header ethernet instanceof ethernet_t
+header ipv4 instanceof ipv4_dddddddddddddddddddddddd0
```
Comment (kamleshbhalui) [inline testdata/p4_16_samples_outputs/pna-too-big-label-name-dpdk.p4.spec:35]: please see here header type modified.

Comment (usha1830) [reply testdata/p4_16_samples_outputs/pna-too-big-label-name-dpdk.p4.spec:35]: So is this addressing the other issue in header instance as well?

Comment (kamleshbhalui) [reply testdata/p4_16_samples_outputs/pna-too-big-label-name-dpdk.p4.spec:35]: no

**Review by psivanup — APPROVED** [review]
Looks fine to me.

**Pull request closed without merging**